src/main/webapp/style.css

```diff
@@ -82,3 +82,72 @@ h1 {
   position: absolute;
   width: 100%;
 }
+
+/* CSS for upload function */
+.bg-model {
+  align-items: center; /* Align vertically*/
+  background: cover;
+  background-attachment: fixed;
+  background-color: rgba(0, 0, 0, 0.9);
+  background-position: center;
+  background-repeat: no-repeat;
+  background-size: cover;
+  color: white;
+  display: none;
+  font-size: 20px;
+  height:90%;
+  justify-content: center;
+  margin-bottom: 3%;
+  margin-top: 4%;
+  position: absolute ;
+  top:0;
+  left:0;
+  padding-bottom: 3rem; /* Padding below the footer */
+  padding-top: 3rem;
+  width: 100%;
+}
+
+.model-content {
+  height: 23%;
+  padding: 10%;
+  position: absolute;
+  top: 0;
+  text-align: center;
+  width: 80%;
+}
+
+input {
+  display:  block;
+  margin: 15px auto;
+  width: 50%;
+}
+
+.close {
+  color: white;
+  cursor: pointer;
+  font-size: 50px;
+  margin: 3%;
+  position: absolute;
+  right: 14px;
+  top: 0;
+  transform: rotate(45deg);
+}
+
+.form-content {
+  padding: 2%;
+  text-align: left;
+}
+
+/* Upload images form is hidden by default. */
+.hidden {
```

Comment:
Shouldn't the `.bg-model` be hidden? At least that's where you toggle hiding and showing.

Reply:
So I added a class '`hidden`' to the `.bg-model` and will remove that class through js after the action link is loaded. It was suggested to do it this way in the example.
What do you think?

Reply:
The .bg-model and the upload image form are hidden differently cause they depend on two different actions. 
The .bg-model depends on the users click while the upload image form depends on blobstore to provide its action url.

Reply:
This is tied to the other comment I had. I think I understand this now.